api/go.mod

```diff
@@ -0,0 +1,22 @@
+module github.com/red-hat-storage/ocs-client-operator/api
+
+go 1.22.0
```

Comment:
Update this to 1.22.5 as well

Reply:
Let's have this update as part of its own commit. The implications on the codebase might be large

Reply:
Updating version here is bumping api-machinery which inturn bumping client-op deps so not to have more than what's required I didn't change version here.